### ISSUE TITLE
Expose `Warning` status for Smoke / CO

### ISSUE THREAD

**Issue by iMicknl** [iMicknl/ha-nest-protect#56]
### Description

Smoke Status is currently a binary sensor (off = OK, on = Emergency). However, Nest Protect also offers a 'Warning' status.

My preference is to keep the binary sensors, since Home Assistant only has the CO/Smoke/Heat device classes for a binary sensor and many automations are based on binary sensors.

Perhaps we can add a separate (text) sensor that shows the state (OK, Warning, Emergency).

**Comment by mdegat01** [comment]
This might be worth an [architecture discussion](https://github.com/home-assistant/architecture) if we can find a few other examples. I'm familiar with it from Nest Protects but I've had other Smoke/CO detectors in the past with the same idea. Like when I've lived in apartment complexes the smoke detectors all had this kind of state transition:

1) Idle - Nothing is wrong
2) Warning - Yours (and only yours) are beeping. Open some windows quick before...
3) Emergency - All detectors in the building go off, everyone out.

I wouldn't surprised if lots of other smoke detectors have this as well. And to your point there really isn't a good way to represent this in HA right now. Your best bad options are a sensor without a device-class and documentation to tell people about the states to expect or multiple binary sensors (one for emergency and one for warning) for each monitored condition.

**Comment by mdegat01** [comment]
I popped onto Zigbee2MQTT for some research since they have that great [supported devices](https://www.zigbee2mqtt.io/supported-devices) page. Obviously not a comprehensive source but they do have a lot of supported devices.

Found a few with a similar idea, a warning attribute. It's a bit broader then this use case but I believe Z2M is doing the aggregation not necessarily the device since it specifies that the attribute is a composite:
- https://www.zigbee2mqtt.io/devices/AV2010_24.html
- https://www.zigbee2mqtt.io/devices/AV2010_24A.html
- https://www.zigbee2mqtt.io/devices/SD-8SCZBS.html

I also noticed what others do is expose a reading of the actual measurement of CO or Smoke. The TuYa and Xiaomi devices I found in these categories seem to take this approach:
- https://www.zigbee2mqtt.io/devices/JTYJ-GD-01LM_BW.html
- https://www.zigbee2mqtt.io/devices/JT-BZ-01AQ_A.html
- https://www.zigbee2mqtt.io/devices/TS0601_smart_CO_air_box.html
- https://www.zigbee2mqtt.io/devices/SD-8SCZBS.html

TuYa and Xiaomi are some pretty popular brands so perhaps that's the more common way? It does work better with HA and wouldn't be the first time Nest took a different approach from others (ex. Battery health vs. Battery level).

Although I actually like Nest's approach better here since it doesn't require me to research what an unsafe level of CO and Smoke is. The battery health thing annoyed me because everyone knows batteries but I have no idea what an unsafe level of CO is (or even what the typical unit of measurement is lol).

**Comment by gminco** [comment]
I like the idea of having the 3 states, what would be the downside of having smoke/co/heat both as binary_sensor and as sensor (with the 3 states)?

**Comment by kdknigga** [comment]
Just out of curiosity, does "warning" currently get lumped in with "ok", or "emergency"?

**Comment by iMicknl** [comment]
> Just out of curiosity, does "warning" currently get lumped in with "ok", or "emergency"?
> 
> 

Currently with 'off', thus 'clear'.

**Comment by kdknigga** [comment]
Thanks!

**Comment by iMicknl** [comment]
Would someone be able to trigger the warning (and emergency status) and gather diagnostics at that point? I assumed that they are using 1 / 2 / 3, however in my latest tests it seems they start at `0`... Would be good to understand if the main functionality of this integration is actually working.

**Comment by kdknigga** [comment]
Just "Download diagnostics" and send the resulting text file after a smoke detection event?

**Comment by iMicknl** [comment]
@kdknigga yes! And if possible, also if you first get a warning message.

**Comment by kdknigga** [comment]
[config_entry-nest_protect-e9c8b4f31b1b8ac2cfccff6f09ca939b_alarm.json.txt](https://github.com/iMicknl/ha-nest-protect/files/8958105/config_entry-nest_protect-e9c8b4f31b1b8ac2cfccff6f09ca939b_alarm.json.txt)
[config_entry-nest_protect-e9c8b4f31b1b8ac2cfccff6f09ca939b_headsup.json.txt](https://github.com/iMicknl/ha-nest-protect/files/8958106/config_entry-nest_protect-e9c8b4f31b1b8ac2cfccff6f09ca939b_headsup.json.txt)

The detector went from "Heads-up" to "Alarm" very quickly with the tester spray, but I think I managed to get diagnostics dumps for each state.  Certainly "smoke_status" is different between the two.

**Comment by willrutherford** [comment]
Any update on this? Having the ability to turn on fans and open windows before reaching an audible alarm state would be excellent!

**Comment by philbert2k** [comment]
> 

Also very interested in this one...

**Comment by IceBotYT** [comment]
Just had my monitoring center call me when the heads up triggered for a bit of smoke coming from the kitchen. I'd rather only have the monitoring center be notified if it goes into Emergency mode and not heads up

**Comment by jessica12ryan** [comment]
I'm also interested in this... I am a developer hobbyist, but I don't have much Python experience. I have a Nest Protect, Home Assistant, and extra hardware and resources if I can be of any help.

**Comment by lowlydba** [comment]
@iMicknl Would you accept a PR doing the simplest version of your original pitch?

> Perhaps we can add a separate (text) sensor that shows the state (OK, Warning, Emergency).

This way we at least have something to consume and build with, and a refactoring later can be done, if deemed worth it.

**Comment by iMicknl** [comment]
@lowlydba yes, we can have a sensor with device_class: enum! Open for contributions. This should be translatable as well.
